### ISSUE TITLE
`use-named-expression` removes comment before the code

### ISSUE THREAD

**Issue by DanielYang59** [sourcery-ai/sourcery#434]
### Checklist

- [x] I have searched the [Sourcery documentation](https://docs.sourcery.ai/) for the issue, and found nothing
- [x] I have read the common issues and fixes listed in the [Sourcery troubleshooting page](https://docs.sourcery.ai/Product/Troubleshooting/#troubleshooting), and found nothing
- [x] I have checked there are no open [bugs](https://github.com/sourcery-ai/sourcery/issues?q=label%3Abug) referencing the same bug or problem

### Description
I believe it's a duplicate of #280 and [mentioned in FAQ](https://docs.sourcery.ai/Coding-Assistant/Product/FAQ/#why-does-sourcery-sometimes-remove-comments), but I didn't see it being resolved in latest version of `sourcery-cli` (not sure when would [a new engine be released](https://github.com/sourcery-ai/sourcery/issues/280#issuecomment-1327275842)). I think comments **above** the code should not have been removed in any case.

### Code snippet that reproduces issue

Noticed this when trying to refactor [the following code for `pymatgen`](https://github.com/materialsproject/pymatgen/blob/b28c9371d684b8a605830cd13060bef9accf68cf/src/pymatgen/analysis/graphs.py#L428-L434).

```bash
sourcery review src/pymatgen/analysis/graphs.py --enable use-named-expression --fix
```

```diff
-        # check we're not trying to add a duplicate edge
-        # there should only ever be at most one edge
-        # between a given (site, jimage) pair and another
-        # (site, jimage) pair
-        existing_edge_data = self.graph.get_edge_data(from_index, to_index)
-        if existing_edge_data:
+        if existing_edge_data := self.graph.get_edge_data(from_index, to_index):
```

### Debug Information

**IDE Version:** 
Visual Studio Code 1.92.1

**Sourcery Version:** 
Sourcery 1.20.0

**Operating system and Version:**
MacOS Sonama 14.6.1

**Comment by alibimahdi** [comment]
Yes this is not good to delete comments. Comments are there for a reason , it would be appreciated that the comments are not deleted by refactoring
